src/OpenTelemetry.AutoInstrumentation/Instrumentations/NLog/Bridge/OpenTelemetryLogHelpers.cs

```diff
@@ -0,0 +1,320 @@
+                        Expression.Constant("arg_"),
+                        Expression.Call(indexVar, typeof(int).GetMethod("ToString", Type.EmptyTypes)!)),
+                    Expression.ArrayIndex(argsParam, indexVar)),
+                Expression.Assign(indexVar, Expression.Add(indexVar, Expression.Constant(1)))),
+            breakLabel);
+
+        return Expression.IfThen(
+            Expression.NotEqual(argsParam, Expression.Constant(null)),
+            Expression.Block(
+                new[] { indexVar },
+                Expression.Assign(indexVar, Expression.Constant(0)),
+                loop));
+    }
+
+    /// <summary>
+    /// Builds the complete EmitLog delegate using expression trees.
+    /// This method constructs a function that can create OpenTelemetry log records
+    /// from NLog event data.
+    /// </summary>
+    /// <param name="logRecordDataType">The LogRecordData type from OpenTelemetry.</param>
+    /// <param name="logRecordAttributesListType">The LogRecordAttributeList type from OpenTelemetry.</param>
+    /// <param name="loggerType">The Logger type from OpenTelemetry.</param>
+    /// <returns>An EmitLog delegate that can create OpenTelemetry log records.</returns>
+    private static EmitLog BuildEmitLog(Type logRecordDataType, Type logRecordAttributesListType, Type loggerType)
+    {
+        // Get the LogRecordSeverity enum type
+        var severityType = logRecordDataType.Assembly.GetType("OpenTelemetry.Logs.LogRecordSeverity")!;
+
+        // Define parameters for the delegate
+        var loggerInstance = Expression.Parameter(typeof(object), "loggerInstance");
+        var body = Expression.Parameter(typeof(string), "body");
+        var timestamp = Expression.Parameter(typeof(DateTime), "timestamp");
+        var severityText = Expression.Parameter(typeof(string), "severityText");
+        var severityLevel = Expression.Parameter(typeof(int), "severityLevel");
+        var exception = Expression.Parameter(typeof(Exception), "exception");
+        var properties = Expression.Parameter(typeof(IEnumerable<KeyValuePair<string, object?>>), "properties");
+        var activity = Expression.Parameter(typeof(Activity), "activity");
+        var args = Expression.Parameter(typeof(object[]), "args");
+        var renderedMessage = Expression.Parameter(typeof(string), "renderedMessage");
+
+        // Build the log record creation expression
+        var logRecordExpression = BuildLogRecord(logRecordDataType, severityType, body, timestamp, severityText, severityLevel, activity);
+
+        // Build the attributes creation expression
+        var attributesExpression = BuildLogRecordAttributes(logRecordAttributesListType, exception, properties, args, renderedMessage);
+
+        // Get the EmitLogRecord method from the logger
+        var emitLogRecordMethod = loggerType.GetMethod("EmitLogRecord", new[] { logRecordDataType, logRecordAttributesListType })!;
```

Comment:
The method in the SDK is [EmitLog](https://github.com/open-telemetry/opentelemetry-dotnet/blob/2a9c406b6da7d10dc9dfbee1f6528e2c85e4e075/src/OpenTelemetry/Logs/LoggerSdk.cs#L27). Is this working correctly?

Reply:
Fixed [665fb09](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/4371/commits/665fb09957bbbe235dd94395d62687b3bf48be33)